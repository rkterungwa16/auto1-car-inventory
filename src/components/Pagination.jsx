import React from 'react'

class Pagination extends React.Component {
  /**
   * skip to the first page
   */
  firstPage () {
    this.props.changePageNumber(1)
    this.props.getPageParams(1)
  }

  /**
   * Skip to the previous page
   */
  previousPage () {
    const {
      currentPage
    } = this.props

    if (currentPage - 1 !== 0) {
      this.props.changePageNumber(currentPage - 1)
      this.props.getPageParams(currentPage - 1)
    } else {
      this.props.changePageNumber(1)
    }
  }

  /**
   * Skip to the last page
   */
  lastPage () {
    const {
      totalPageCount
    } = this.props

    this.props.changePageNumber(totalPageCount)
    this.props.getPageParams(totalPageCount)
  }

  /**
   * Skip to next page
   */
  nextPage () {
    const {
      totalPageCount,
      currentPage
    } = this.props

    if (currentPage + 1 > totalPageCount) {
      this.props.changePageNumber(totalPageCount)
      this.props.getPageParams(totalPageCount)
    } else {
      this.props.changePageNumber(currentPage + 1)
      this.props.getPageParams(currentPage + 1)
    }
  }
  render () {
    const {
      currentPage,
      totalPageCount
    } = this.props

    return (
      <div className='pagination'>
        <a
          href='#view details'
          className='available-cars__text available-cars__text--small link__text--orange'
          onClick={this.firstPage.bind(this)}
        >First</a>
        <a
          href='#view details'
          className='available-cars__text available-cars__text--small link__text--orange'
          onClick={this.previousPage.bind(this)}
        >Previous</a>
        <a
          href='#view details'
          className='available-cars__text available-cars__text--small link__text--black'
        >{`Page ${currentPage} of ${totalPageCount}`}
        </a>
        <a
          href='#view details'
          className='available-cars__text available-cars__text--small link__text--orange'
          onClick={this.nextPage.bind(this, currentPage, totalPageCount)}
        >Next</a>
        <a
          href='#view details'
          className='available-cars__text available-cars__text--small link__text--orange'
          onClick={this.lastPage.bind(this)}
        >Last</a>
      </div>
    )
  }
}

Pagination.defaultProps = {
  currentPage: 1
}

export default Pagination
