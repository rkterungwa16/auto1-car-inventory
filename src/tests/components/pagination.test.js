import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Pagination from '../../components/Pagination'

Enzyme.configure({ adapter: new Adapter() })
const { shallow } = Enzyme

const getPageParams = jest.fn()
const wrapper = shallow(
  <Pagination
    getPageParams={getPageParams}
    totalPageCount={100}
  />
)
test('should render Pagination correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should render Pagination with alt data correctly', () => {
  wrapper.setProps({
    totalPageCount: 5
  })
  expect(wrapper).toMatchSnapshot()
})
