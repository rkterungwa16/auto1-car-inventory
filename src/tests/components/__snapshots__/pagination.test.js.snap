// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Pagination correctly 1`] = `
<div
  className="pagination"
>
  <a
    className="available-cars__text available-cars__text--small link__text--orange"
    href="#view details"
    onClick={[Function]}
  >
    First
  </a>
  <a
    className="available-cars__text available-cars__text--small link__text--orange"
    href="#view details"
    onClick={[Function]}
  >
    Previous
  </a>
  <a
    className="available-cars__text available-cars__text--small link__text--black"
    href="#view details"
  >
    Page 1 of 100
  </a>
  <a
    className="available-cars__text available-cars__text--small link__text--orange"
    href="#view details"
    onClick={[Function]}
  >
    Next
  </a>
  <a
    className="available-cars__text available-cars__text--small link__text--orange"
    href="#view details"
    onClick={[Function]}
  >
    Last
  </a>
</div>
`;

exports[`should render Pagination with alt data correctly 1`] = `
<div
  className="pagination"
>
  <a
    className="available-cars__text available-cars__text--small link__text--orange"
    href="#view details"
    onClick={[Function]}
  >
    First
  </a>
  <a
    className="available-cars__text available-cars__text--small link__text--orange"
    href="#view details"
    onClick={[Function]}
  >
    Previous
  </a>
  <a
    className="available-cars__text available-cars__text--small link__text--black"
    href="#view details"
  >
    Page 1 of 5
  </a>
  <a
    className="available-cars__text available-cars__text--small link__text--orange"
    href="#view details"
    onClick={[Function]}
  >
    Next
  </a>
  <a
    className="available-cars__text available-cars__text--small link__text--orange"
    href="#view details"
    onClick={[Function]}
  >
    Last
  </a>
</div>
`;
